app/workers/dependency_resolution_worker.rb

```diff
@@ -41,6 +36,20 @@ def assign_attributes(args)
     @source_fields = args.fetch(:source_fields, [])
   end
 
+  def orphaned_content_ids_for_locale
+    Document
+      .distinct
```

Comment:
Can you explain why you're using `distinct` here? I've done some searching and it looks like `.distinct` is [usually applied _after_ a query](https://api.rubyonrails.org/v6.1.0/classes/ActiveRecord/QueryMethods.html#method-i-distinct). I'm sure it's just something I've not come across before :)

Reply:
Sure thing. I'm using `distinct` so that there are no duplicates returned in the result set (`distinct` being ActiveRecord's answer to [SQL distinct](https://www.w3schools.com/sql/sql_distinct.asp)). I noticed duplicate content ids may be returned if a live and draft edition are present for a document when querying for the draft content store, e.g. ` .where(editions: { content_store: %w[draft live] }`.

As for placement of the mighty distinct. I usually place it at the start, probably just out of habit much the same as when you write a `DISTINCT` in SQL it's added at the start of the query.

Hope that answers your question 😄

Reply:
It does - thanks!